src/Core/Managed/Shared/Channel/Transmission.cs

```diff
@@ -221,6 +238,38 @@ protected virtual WebRequest CreateRequest(Uri address)
             return request;
         }
 
+#if NET40
+        private Task<HttpWebResponseWrapper> SendRequestAsync(WebRequest request)
+        {
+            return Task.Factory.FromAsync(request.BeginGetRequestStream, request.EndGetRequestStream, null)
+                .ContinueWith(
+                    task =>
+                    {
+                        var requestStream = task.Result;
+                        return Task.Factory.FromAsync(
+                            (callback, o) => requestStream.BeginWrite(Content, 0, Content.Length, callback, o),
+                            requestStream.EndWrite, 
+                            null).ContinueWith(
+                                task1 =>
+                                {
+                                    requestStream.Dispose();
+                                    task1.RethrowIfFailed();
+                                });
+                    }).Unwrap().ContinueWith(task =>
+                    {
+                        task.RethrowIfFailed();
```

Comment:
You know that the task is completed. You could get the exception by asking for the result and unwrapping the aggregate (to be like await). Maybe adjust the implementation of RethrowIfFailed?